Private/Find-AuditingIssue.ps1

```diff
@@ -8,26 +8,22 @@
         ($_.objectClass -eq 'pKIEnrollmentService') -and
         ($_.AuditFilter -ne '127')
     } | ForEach-Object {
-        $Issue = New-Object -TypeName pscustomobject
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Forest' -Value $_.CanonicalName.split('/')[0] -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Name' -Value $_.Name -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'DistinguishedName' -Value $_.DistinguishedName -Force
+        $Issue = [pscustomobject]@{
+            Forest            = $_.CanonicalName.split('/')[0]
+            Name              = $_.Name
+            DistinguishedName = $_.DistinguishedName
+            Technique         = 'DETECT'
+        }
         if ($_.AuditFilter -match 'CA Unavailable') {
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Issue' -Value $_.AuditFilter -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Fix' -Value 'N/A' -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Revert' -Value 'N/A' -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Technique' -Value 'DETECT' -Force
+            $Issue['Issue']  = $_.AuditFilter
+            $Issue['Fix']    = 'N/A'
+            $Issue['Revert'] = 'N/A'
         }
         else {
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Issue' -Value "Auditing is not fully enabled on $($_.CAFullName). Current value is $($_.AuditFilter)" -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Fix' `
-                -Value "certutil.exe -config `'$($_.CAFullname)`' -setreg `'CA\AuditFilter`' 127; Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }" -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Revert' `
-                -Value "certutil.exe -config $($_.CAFullname) -setreg CA\AuditFilter  $($_.AuditFilter); Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }" -Force
-            $Issue | Add-Member -MemberType NoteProperty -Name 'Technique' -Value 'DETECT' -Force
+            $Issue['Issue']  = "Auditing is not fully enabled on $($_.CAFullName). Current value is $($_.AuditFilter)"
+            $Issue['Fix']    = "certutil.exe -config `'$($_.CAFullname)`' -setreg `'CA\AuditFilter`' 127; Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }"
+            $Issue['Revert'] = "certutil.exe -config $($_.CAFullname) -setreg CA\AuditFilter  $($_.AuditFilter); Invoke-Command -ComputerName `'$($_.dNSHostName)`' -ScriptBlock { Get-Service -Name `'certsvc`' | Restart-Service -Force }"
         }
-        $Severity = Set-Severity -Issue $Issue
```

Comment:
I don't see severity being assigned in the new version. Did you move it?

Reply:
We never actually used the Severity function, so I removed it.